### PR TITLE
When you click a system in the map mission view, iterate through all missions that involve the system.

### PR DESCRIPTION
**Pull request by flaviojs**
This allows you to know which of the missions has the system as a waypoint or has a system planet to land on.

### CODE REVIEW

**Comment by jafdy** [comment]
Screen shot?

**Comment by flaviojs** [comment]
There's nothing to screenshot? When you click a system:

* old behavior = select next accepted or available mission with that destination (skips missions that have a waypoint or stopover key there) 
* new behavior = old behavior missions + missions with a waypoint or stopover in the system

**Comment by jafdy** [comment]
I read this as "highlight all missions that involve that system", as in multiple lines get highlighted. If that's not what it is then don't mind me.

File: source/MissionPanel.cpp [hunk 1]
```diff
@@ -43,6 +43,26 @@ using namespace std;
 
 namespace {
 	const int SIDE_WIDTH = 280;
+	
+	// Checks if a missions involves the provided system.
```
Comment (tehhowch) [inline source/MissionPanel.cpp]: `Determine if the given mission uses the given system`

or at least `a` -> `this` and drop the s on `missions`

File: source/MissionPanel.cpp [hunk 2]
```diff
@@ -43,6 +43,26 @@ using namespace std;
...
+	bool Involves(const Mission &mission, const System *system)
+	{
+		if(system == nullptr)
```
Comment (tehhowch) [inline source/MissionPanel.cpp]: !system

**Comment by flaviojs** [comment]
Hmm, the function fits better as part of the Mission class but I can't think of another place that would use it.

**Comment by tehhowch** [comment]
I was thinking similarly too. There may be a use for a Mission::LandingTarget(const System *) function that unions its stopovers and destination in the given system, then returns a(n accessible) stellarobject pointer for the first result (preferring an unvisited stopover to the destination).
This function would replace two of your steps here (you'd just test for a nullptr return), and could be used in AI::MovePlayer as well.

**Comment by tehhowch** [comment]
The construct I mention above could be used to achieve #1452 - when landing, do not roll for illegal items if this planet is associated with a mission that gives clearance. Probably best done an anonymous namespace method that takes a missions list and a planet pointer, and checks to see if any mission gives clearance for that planet

**Comment by flaviojs** [comment]
A waypoint only implies entering the system, no landing target/planet, so it wouldn't replace my function. Since refactoring other code is out of scope for this PR, I'll leave it as is unless MZ tells me to move my function to Mission.

**Comment by tehhowch** [comment]
Indeed, it would not. It would only handle the destination and stopover portions.

**Pull request merged**